modules/nf-core/admixture/main.nf

```diff
@@ -0,0 +1,43 @@
+
+process ADMIXTURE {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "bioconda::admixture=1.3.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/admixture:1.3.0--0':
+        'quay.io/biocontainers/admixture:1.3.0--0' }"
+
+    input:
+    tuple val(meta), path(bed), path(bim), path(fam)
```

Comment:
You could get creative with `stageAs` here to stage the different output types in separate directories?

Reply:
and then use the dir name is target for `$input_file`

```
input_file = bed ? "bed_dir/*.bed" : ped_or_geno ? "ped_dir/*.{ped,geno}": error "No bed or ped/geno given"

```
(code untested)